### PR TITLE
Migrate K8s Gateway API to v1beta1 version

### LINKED ISSUE #5797: K8s GW API Support  - v1alpha2 is deprecated
Currently "gateway.networking.k8s.io/v1alpha2" is deprecated, K8s GW API Kiali support and yaml test examples should be switched to "gateway.networking.k8s.io/v1beta1"

### PR DESCRIPTION
**Pull request by ferhoyos**
Fixes #5797

### CODE REVIEW

File: kubernetes/istio.go [hunk 1]
```diff
@@ -507,8 +507,8 @@ func ParseRegistryConfig(config map[string][]byte) (*RegistryConfiguration, erro
 							registry.EnvoyFilters = append(registry.EnvoyFilters, ef)
 						case "Gateway":
 							// It needs to figure out Gateway object type by apiVersion, whether it is Gateway API of Istio Gateway
-							if mItem["apiVersion"] == K8sApiNetworkingVersionV1Beta1 || mItem["apiVersion"] == K8sApiNetworkingVersionV1Alpha2 {
```
Comment (hhovsepy) [inline kubernetes/istio.go:510]: I think we should leave it as backward compatible, to list existing v1alpha2 Gateways.

**Comment by hhovsepy** [comment]
Newly created K8s v1beta1 Gateway objects are not listed.

File: kubernetes/istio.go (hunk 1 above)
Comment (ferhoyos) [reply kubernetes/istio.go:510]: done

**Comment by hhovsepy** [comment]
works fine for me now, will approve after tests pass, thank you @ferhoyos

**Review by hhovsepy — APPROVED** [review]

**Review by aljesusg — APPROVED** [review]
LGFM

**Pull request merged**